cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -508,6 +509,38 @@ pub trait ClientTrait {
     ) -> Result<HashMap<String, Vec<StoreInfo>>, CatalogClientError>;
 }
 
+/// Wrapper around `semver::VersionReq` which attempts to normalize semver
+/// strings in a format that the catalog supports.
+#[derive(Debug, Error)]
+struct SemverNormalized(VersionReq);
+
+impl TryFrom<String> for SemverNormalized {
```

Comment:
suggestion: use `FromStr` trait instead.
Just because its the more common trait for string parsing, and we're specifically calling the `FromStr` impl of `VersionReq` in the implementation, it seems odd to use a different trait here.

(fwiw that woudl also allow us to use drop the clone in https://github.com/flox/flox/pull/3082/files#diff-876ee899124e7eb1c87d3ed6f8c55ed4a19ca763a47639e551e4e1877798e868R562)